### PR TITLE
Add-on information for onboarding premium users on mobile

### PR DESCRIPTION
**Pull request by codemist**
This adds information about the add-on for premium users onboarding on mobile. This PR also updated the strings in the onboarding flow, and some corrected some UI inconsistencies as per the Figma spec. For the third step, the copy switches from a reply/extension value prop depending on if a user is on mobile or desktop.

# Screenshot (if applicable)
image

# How to test

Make the `premium-onboarding` class visible by unselecting the `display: none;` property, add `is-visible` utility class to  `c-premium-onboarding-step-3`. Alternatively, make a new account, upgrade to premium and walk through onboarding steps.

# Checklist

- [x] l10n dependencies have been merged, if any.
- [x] All acceptance criteria are met.
- [x] All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/static/scss/libs/protocol/css/includes/tokens/dist/index.scss`).
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: privaterelay/templates/includes/premium-onboarding.html
Comment (Vinnl) [inline privaterelay/templates/includes/premium-onboarding.html:84]: Blocking: I think what we want to do here is show the new strings on small screens, but the old strings on larger screens. This is because on larger screens, we show a "Get Relay Extension" button, so text about getting the extension makes sense, whereas on smaller screens, we don't, because the extension is not available there.

**Review by Vinnl — CHANGES_REQUESTED** [review]
I added some suggestions, but most importantly one blocking change: I think we'll still want to promote the extension on large screens.

File: privaterelay/templates/includes/premium-onboarding.html
Comment (codemist) [reply privaterelay/templates/includes/premium-onboarding.html]: Ah, that was an oversight on my part. Thanks for the catch - just filed a new PR for that localization error here: https://github.com/mozilla-l10n/fx-private-relay-l10n/pull/50

File: privaterelay/templates/includes/premium-onboarding.html
Comment (codemist) [reply privaterelay/templates/includes/premium-onboarding.html]: Removed all Eng alt messages. It was there before I made this PR so I wasn't sure if I needed to touch it, but it makes sense to remove it entirely.

File: privaterelay/templates/includes/premium-onboarding.html
Comment (codemist) [reply privaterelay/templates/includes/premium-onboarding.html:88]: Updated to `profile-label-continue`. Probably a minor issue for now since we can always just swap it out for the old string, but I'll ask them for their recommendation.

File: privaterelay/templates/includes/premium-onboarding.html
Comment (codemist) [reply privaterelay/templates/includes/premium-onboarding.html]: Blocking this for now until that PR gets merged in.

File: privaterelay/templates/includes/premium-onboarding.html
Comment (codemist) [reply privaterelay/templates/includes/premium-onboarding.html]: Good call! I added a mobile/desktop condition to show different types of copy.

https://user-images.githubusercontent.com/13066134/147238298-95280e9d-1370-4838-85d8-d25a9b904290.mov

Comment (codemist) [reply privaterelay/templates/includes/premium-onboarding.html]: https://user-images.githubusercontent.com/13066134/147238552-04ba4a44-9fd6-4ebb-8b15-3d91b861c959.mov

File: static/scss/partials/onboarding.scss
Comment (codemist) [reply static/scss/partials/onboarding.scss]: Agh, jet lagged coding brain things. Yeah, I was supposed to remove that! Good catch.

Comment (codemist) [reply static/scss/partials/onboarding.scss]: Just wondering what your best approach is for hard overriding certain attributes? I know we have a `utils` file with a bunch of overriding classes, in the future if we must override for whatever reason do you feel that adding it in that file is good practice?

Comment (Vinnl) [reply static/scss/partials/onboarding.scss]: The most important strategy is trying to avoid writing selectors that people will want to override, so that's outside of the scope of this PR, but basically it comes down to avoid reuse of classes unless you're almost 100% certain that the two use cases are exactly the same in every way and will remain so even in future design iterations. Which is rarely the case.

There are tools that can help with this that I'd like us to consider at some point, e.g. [CSS modules](https://css-tricks.com/css-modules-part-1-need/) (not to be confused with [the other CSS modules](https://css-tricks.com/css-modules-the-native-ones/), [Tailwind](https://tailwindcss.com/), and others. I've got a proposal for this that I'm hoping we'll look at in January.

Of course, that doesn't really help us _today_, where we have a lot of (overly, IMHO) generic selectors, some even just styling elements in general, rather than with specific classes. In that case, I look at the selector that's adding styles I don't like, and make sure that mine is more specific. For example, say we have:

```html
<div class="my-class">
  <h2 class="generic-header">
</div>
```

and

```css
/* Somewhere in Protocol */
h2 {
  line-height: 2;
}

/* our_style.css */
.my-class {
  line-height: 1.5;
}
```

Then I can make sure it applies to that h2 by using a more [specific](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity) selector:

```css
.my-class h2 {
  line-height: 1.5;
}
```

(Ideally I'd be able to update the original `h2` styling to use a more specific selector that has to be explicitly opted into, but since I don't know every `h2` it's supposed to match with, just using ever more specific selectors is about the best we can do. Which is why people sometimes refer to CSS as an "append-only" language: it's really hard to update/remove existing CSS.)

File: privaterelay/templates/includes/premium-onboarding.html
Comment (Vinnl) [inline privaterelay/templates/includes/premium-onboarding.html:21]: I just realised that `multi-part-onboarding-premium-welcome-description` says

> On desktop, you can use the Relay add-on to create them on the go.

So maybe here, too, we need to display different content depending on the screen size?

File: privaterelay/templates/includes/premium-onboarding.html
Comment (Vinnl) [inline privaterelay/templates/includes/premium-onboarding.html:74]: I think this headline should probably also differ between small and large screens - it currently says "Reply to your emails" on large screens, rather than something about extensions.

**Review by Vinnl — APPROVED** [review]
Cool! Oh also, I use another way to test this, which might be interesting to you as well: I go to http://127.0.0.1:8000/admin/emails/profile/, then set "Onboarding state" to `0` for the profile I'm testing with.

Two more small suggestions inline, other than that, once that l10n string is in, this is good to go!

File: privaterelay/templates/includes/premium-onboarding.html
Comment (codemist) [reply privaterelay/templates/includes/premium-onboarding.html]: Done!

File: privaterelay/templates/includes/premium-onboarding.html
Comment (codemist) [reply privaterelay/templates/includes/premium-onboarding.html:21]: Ohhh cool, I had no idea I could use admin for that. Solves so many testing hurdles! Lol.

File: static/scss/partials/onboarding.scss
Comment (codemist) [reply static/scss/partials/onboarding.scss]: Got it, I've never even heard of CSS Modules, so that was a really interesting read. Thank you! Using specific selectors is definitely super helpful (hence why I really appreciate nesting in SASS).

File: privaterelay/templates/includes/premium-onboarding.html
Comment (codemist) [reply privaterelay/templates/includes/premium-onboarding.html]: No longer blocked.

**Pull request merged**
